src/recordlinker/linking/mpi_service.py

```diff
@@ -27,12 +27,10 @@ def get_block_data(
     # multiple times, once for each Blocking Key.  If a Patient record
     # has a matching Blocking Value for all the Blocking Keys, then it
     # is considered a match.
-    for idx, block in enumerate(algo_config["blocks"]):
-        key_name = block["value"].upper()
-        if not hasattr(models.BlockingKey, key_name):
-            raise ValueError(f"Invalid Blocking Key: {block}")
-        # Get the matching Blocking Key based on the value in the algo_config
-        key = models.BlockingKey[key_name]
+    for idx, key_id in enumerate(algorithm_pass.blocking_keys):
+        #get the BlockingKey obj from the id
+        key = models.BlockingKey.from_id(key_id)
```

Comment:
I think the `if not hasattr(models.BlockingKey, key_name)` check is still a good one here.  Because the data is coming from the model, we should be verifying that the data is valid before persisting now, so at this point in time it really can't be a user error.  But its possible that data is corrupted or some bug is introduced, so an assertion would be helpful here just in case.
```suggestion
        assert hasattr(models.BlockingKey, key_id), "Invalid Blocking Key: {key_id}")
        key = models.BlockingKey.from_id(key_id)
```

Reply:
I added a `ValueError` in the `models.BlockingKey.from_id(...)` function that raises if there's no key with the given `key_id`. I was considering adding this assert but I don't think it will recognize the `key_id` as a attribute since its just an `int` and not an actual `BlockingKey` object

Reply:
I can go ahead and change `AlgorithmPass` model to store a `list[str]` like you commented about below, then I could remove the `from_id(...)` function and just do the assert statement you suggested above